app/controllers/users_controller.rb

```diff
@@ -94,7 +94,22 @@ def remove_association
   end
 
   def onboarding_update
-    current_user.update(JSON.parse(params[:user]).to_h) if params[:user]
+    current_user.update(params[:user].permit(:summary, :location, :employment_title, :employer_name)) if params[:user]
```

Comment:
Since the code is calling `save!` anyway afterwards, maybe there's no need to update the database twice, you can probably use [assign_attributes](https://api.rubyonrails.org/classes/ActiveModel/AttributeAssignment.html#method-i-assign_attributes) here - I might be wrong in case `authorize User` needs something to be persisted